src/backend/nodes/readfast.c

```diff
@@ -892,7 +909,7 @@ _readOidAssignment(void)
 	READ_LOCALS(OidAssignment);
 
 	READ_OID_FIELD(catalog);
-	READ_STRING_FIELD(objname);
+	READ_STRING_FIELD_NULL(objname);
```

Comment:
The read doesn't match the write in https://github.com/cloudberrydb/cloudberrydb/blob/main/src/backend/nodes/outfuncs_common.c#L323

Reply:
In here, it process when the objname is '', which is just a '\0' in memory. In output, it will add '\0' in appendBinaryStringInfo which is correct while in input, it will replace '\0' to a NULL pointer, which is not the same meaning.

Reply:
In here, it process when the objname is '', which is just a '\0' in memory. In output, it will add '\0' in appendBinaryStringInfo which is correct while in input, it will replace '\0' to a NULL pointer, which is not the same meaning.